### ISSUE TITLE
train on youtube-vos

### ISSUE THREAD

**Issue by colorblank** [visinf/dense-ulearn-vos#2]
Grear work! Thanks for sharing your code!

I use the default training configure of ytvos to train the network.
But I only got best performance J&F=65.5 at epoch 490.  
Is the default configure supposed to have this performance? 
How can I get the best performance like your provided checkpoint?
I'd appreciate it if you could point out what I did wrong.
![image](https://user-images.githubusercontent.com/16894837/142725521-8afb93d5-48e0-4af7-ba79-d46f0f552cb9.png)

**Comment by arnike** [comment]
Hi @colorblank,
thanks! Are you referring to "0.655" in the snapshot name, or did you actually use `infer_vos.sh` script to evaluate on the whole validation set? Please note that we save the snapshot based on only 5 sequences from DAVIS-2017 during training, so the accuracy on the complete validation set will be different and tends to be higher.
Best,
Nikita

**Comment by colorblank** [comment]
My bad. I tried to the test code and got th results like:
```
J&F-Mean   J-Mean  J-Recall  J-Decay   F-Mean  F-Recall  F-Decay
 0.680324 0.658446  0.789883 0.182041 0.702202  0.835232 0.258544
```
Thanks for your clarification.
Besides, there is a small bug in this line : https://github.com/visinf/dense-ulearn-vos/blob/9e39d359a53a2343522ce5820fdf27223a4ffcb4/infer_vos.py#L327. You should change to single gpu mode like "model = model.cuda()".

**Comment by arnike** [comment]
Thanks for the pointer about the inference code.

There may be some deviation in accuracy between the runs, albeit your results are well below what I typically observed. You can try simply re-runing the training and/or decreasing the interval between the validation runs (check `LOG.ITER_VAL` in the YAML config).

I've now also experimented with decaying the learning rate for training on YouTube-VOS to make it more reproducible. My only change to the default setup was in `config/ytvos.yaml`:
```
MODEL.LR_SCHEDULER: "step"
MODEL.LR_GAMMA: 0.5
MODEL.LR_STEP: 100
```
This is the result I got on the complete DAVIS-2017 val:
```
--------------------------- Global results for val ---------------------------
 J&F-Mean    J-Mean  J-Recall   J-Decay    F-Mean  F-Recall   F-Decay
 0.694455  0.672305  0.818764  0.164835  0.716605  0.843103  0.239955
```
(The maximum on the validation subset we use in training was 64.5 J&F  reached at epoch 330.)
Let me know if that works for you.
Best regards,
Nikita

**Comment by JerryX1110** [comment]
@colorblank Hi, I'd like to know how long does it take to train on YTB, thanks!
